sw/otbn/code-snippets/p256.s

```diff
@@ -294,98 +212,84 @@ mod_mul_256x256:
  *                               x-coordinate of input point
  * @param[in]  dmem[24]: dptr_y, pointer to dmem location containing affine
  *                               y-coordinate of input point
- * @param[in]  w27: b, curve domain parameter
- * @param[in]  w29: p, modulus, 2^256 > p > 2^255.
- * @param[in]  w28: u, pre-computed Barrett constant (without u[256]/MSb
- *                       of u which is always 1 for the allowed range.
- * @param[in]  w31: all-zero.
- * @param[in]  MOD: p, modulus, 2^256 > p > 2^255.
  *
- * clobbered registers: x8, x13, x14, x19, x20, x21, w0, w19 to w25
+ * clobbered registers: x2, x3, x19, x20, w0, w19 to w25
  * clobbered flag groups: FG0
  */
 p256_isoncurve:
 
-  addi      x3, x0, 0
-  bn.lid    x3, 0(x0)
-
-  lw        x16, 0(x0)
-  lw        x17, 4(x0)
-  lw        x18, 8(x0)
-
-  /* load dmem pointer to signature r from dmem: x19 <= dptr_r = dmem[12] */
-  lw        x19, 12(x0)
-
-  /* load dmem pointer to signature s from dmem, x20 <= dptr_s = dmem[16] */
-  lw        x20, 16(x0)
-
-  /* load dmem pointer to affine x-coordinate of curve point from dmem:
-     x21 <= dptr_x = dmem[20] */
-  lw        x21, 20(x0)
-
-  /* load dmem pointer to affine y-coordinate of curve point from dmem:
-     x22 <= dptr_y = dmem[24] */
-  lw        x22, 24(x0)
-
-  lw        x23, 28(x0)
+  /* setup all-zero reg */
+  bn.xor    w31, w31, w31
 
-  /* setup pointers to WREGs */
-  addi      x8, x0, 0
+  /* load dmem pointer to signature r in dmem: x19 <= dptr_r = dmem[12] */
```

Comment:
I'd leave out the `dmem[12]` part here and similarly below, since it really depends on what the use case and what the linker happens to do (grep for it, there are a number of them in the source code)

Reply:
Actually we use these kind of comments a lot, also in the documentation headers of the externally callable routines in both RSA and ECC lib. So if we change this, we should change it everywhere. My understanding and assumption here so far was that the first dmem word will always contain the index table. And that the respective symbols might be exported in the .elf but also that calling conventions would be well-defined if the OTBN binary is provided in a non-elf format. WDYT?

Reply:
Ah I see. Let's just leave it then.